gsa/src/web/components/dialog/__tests__/multistepfooter.js

```diff
@@ -0,0 +1,180 @@
+/* Copyright (C) 2018 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import React from 'react';
+
+import {render, fireEvent} from 'web/utils/testing';
+import Theme from 'web/utils/theme';
+
+import MultiStepFooter from 'web/components/dialog/multistepfooter';
+import usePageTurn from 'web/components/hooks/usepageturn';
+
+const lastPage = 2;
+// eslint-disable-next-line react/prop-types
+const DummyPage = ({customHook}) => {
+  // hooks MUST be called inside functions
+  const {handlePageTurn, prevDisabled, nextDisabled} = customHook(lastPage);
+
+  return (
+    <MultiStepFooter
+      prevDisabled={prevDisabled}
+      nextDisabled={nextDisabled}
+      rightButtonTitle={'Save'}
+      onNextButtonClick={() => handlePageTurn('Next')}
+      onPreviousButtonClick={() => handlePageTurn('Previous')}
+    />
+  );
+};
+
+describe('MultiStepFooter tests', () => {
+  test('should render', () => {
+    const {element} = render(<MultiStepFooter rightButtonTitle="Foo" />);
+
+    expect(element).toMatchSnapshot();
+
+    const button = element.querySelector('button[title="Foo"]');
+
+    expect(button).toHaveStyleRule('background', Theme.lightGreen);
+  });
+
+  test('should render loading and disable cancel button', () => {
+    const {element} = render(
+      <MultiStepFooter rightButtonTitle="Foo" loading={true} />,
+    );
+
+    expect(element).toMatchSnapshot();
+
+    const button = element.querySelector('button[title="Foo"]');
```

Comment:
When using English strings it might be necessary to set the locale to 'en' to make sure the test will pass on all machines independent of the language setting

Reply:
Done :)